cogs/reward.py

```diff
@@ -4,6 +4,95 @@
+                
+        def spellEmbedCheck(r, u):
+            sameMessage = False
+            if charEmbedmsg.id == r.message.id:
+                sameMessage = True
+            return sameMessage and ((r.emoji in alphaEmojis[:6]) or (str(r.emoji) == '❌')) and u == author
+            
+        #Puts all rewards into an array
+        for i in range(0,int(amount)):
+            if not isinstance(randomItem[i]['Name'], str): #If one of the items has subchoices, such as ammunition, only the category will be added instead of an array of all the choices
+                temp = sample(randomItem[i]['Name'], 1)[0]
+                tempstr = str(temp)
+                rewardString.append(tempstr)
+            elif 'Spell Scroll' in randomItem[i]['Name']:
+                spell = re.findall(r"\d+", randomItem[i]['Name'])
+                
+                # create an embed object for user communication
+                #extract spell level:
+                spellLevel = int(spell[0])
+                charEmbed = discord.Embed()
+                charEmbed.title = f"{randomItem[i]['Name']}"
+                charEmbed.description = f"What class list would you like the spell scroll to be from?"
+                
+                #Determine if the scroll level is available to half-casters
+                if spellLevel < 6: #all caster list
+                    spellClasses = ["Artificer", "Bard", "Cleric", "Druid", "Ranger", "Paladin", "Sorcerer", "Warlock", "Wizard"]
+                    charEmbed.add_field(name="Please pick the spell list you want the scroll to be from:", value=f"{alphaEmojis[0]}: Artificer\n{alphaEmojis[1]}: Bard\n{alphaEmojis[2]}: Cleric\n{alphaEmojis[3]}: Druid\n{alphaEmojis[4]}: Ranger\n{alphaEmojis[5]}: Paladin\n{alphaEmojis[6]}: Sorcerer\n{alphaEmojis[7]}: Warlock\n{alphaEmojis[8]}: Wizard\n", inline=False)
+                else: #full caster list
+                    spellClasses = ["Bard", "Cleric", "Druid", "Sorcerer", "Warlock", "Wizard"]
+                    charEmbed.add_field(name="Please pick the spell list you want the scroll to be from:", value=f"{alphaEmojis[0]}: Bard\n{alphaEmojis[1]}: Cleric\n{alphaEmojis[2]}: Druid\n{alphaEmojis[3]}: Sorcerer\n{alphaEmojis[4]}: Warlock\n{alphaEmojis[5]}: Wizard\n", inline=False)
+                    
+                charEmbedmsg = await channel.send(embed=charEmbed, content="Testing")
+                await charEmbedmsg.add_reaction('❌')
+                
+                try:
+                    await charEmbedmsg.edit(embed=charEmbed)
+                    await charEmbedmsg.add_reaction('❌')
+                    tReaction, tUser = await self.bot.wait_for("reaction_add", check=spellEmbedCheck, timeout=60)
+                except asyncio.TimeoutError:
+                    await charEmbedmsg.delete()
+                    await channel.send(f'Spell list selection timed out! Try again using the same command:\n```yaml\n{commandPrefix}{tier}{rewardType}{amount}```')
```

Comment:
The last part of the message, is that your debugging info? If yes delete it here and below